### ISSUE TITLE
Data API

### ISSUE THREAD

**Issue by muellerzr** [Novetta/adaptnlp#129]
We *probably* should have a data API of some form, that ties into https://github.com/Novetta/adaptnlp/issues/128

Ideally it should simply prep a dataset for tokenization of a model, or tokenize the data itself.

For now we cover two inputs:
1. Individual texts
2. CSV

We should support something akin to `fastai`'s `get_y`, but with decent defaults so that customization *is* available, but not needed.

Ideally something like:
```python
dset = TaskDataset.from_df(
  df,  # Can be fname or dataframe
  get_x = ColReader('text'),
  get_y = ColReader('label'),
  splitter = RandomSplitter(),
  model = 'bert-base-uncased', # The name/type of downstream model
  task = "ner" # Or use a `Task.NER` namespace class
)
```

And further:
```python
dset.dataloaders(bs=8, collate_fn=data_collator)
```

It reads *extremely* similar to the fastai API, but we do not use the fastai API, as for text doing it like this is a bit easier.

The highest level API would look like so:
```python
dls = TaskDataLoaders.from_df(df, 'text', 'label', model='bert-base-uncased')
```
We should note the model used, and when integrating it with the tuning API if something is off with the model entered, we make note of that

**Comment by muellerzr** [comment]
There should be *extremely* detailed tutorials about how to write your own `get_x` and `get_y`, along with `splitter`, so for those who are not familiar with fastai, they can write their own

**Comment by muellerzr** [comment]
Closed with the release of 0.3.0
